src/com/game/controller/Game.java

```diff
@@ -0,0 +1,133 @@
+/*
+    This is the game class. It instantiates all of the games materials, and updates them. The game will use the model package and view package to pass data from the user to the system and vice versa.
+ */
+package com.game.controller;
+
+
+import com.game.client.Client;
+import com.game.model.engine.LogicEngine;
+import com.game.model.materials.Caterpillar;
+import com.game.model.materials.Enemy;
+import com.game.model.materials.Location;
+import com.game.view.ViewWindow;
+import com.sun.tools.javac.Main;
+
+import java.io.*;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Scanner;
+
+public class Game {
+
+    private HashMap<String, Location> locations;
+    private HashMap<String, Enemy> enemies;
+    public static Caterpillar caterpillar;
+    private static LogicEngine processor;
+    private ViewWindow viewWindow;
+
+    public static LogicEngine getProcessor() {
+        return processor;
+    }
+
+    public void setProcessor(LogicEngine processor) {
+        this.processor = processor;
+    }
+
+    public Game() {
+
+    }
+    //This should be called by the client to start a new game.
+    public void start(){
+        setUpComponents();
+//      Hongyi: correct autometic health increase and move the refresh window function to input panel
+//        run();
+    }
+
+    //This method is designed to instantiate the necessary fields of a Game object.
+    private void setUpComponents() {
+        this.enemies = populateEnemies();
+        this.locations = populateLocations();
+        this.caterpillar = new Caterpillar(100,0,0);
+        this.processor = new LogicEngine(caterpillar,locations, enemies);
+        this.caterpillar.setCurrentLocation(locations.get("GENESIS"));
+        this.viewWindow = new ViewWindow();
+    }
+    //This class controls the game loop. As the user inputs information the view will be updated.
+    //I want an instructions panel to be read and you cant start the game until you hit
+    private void run() throws IOException {
+        int counter = 0;
+        while (true){
+            viewWindow.updateCaterpillarStatus();
+            caterpillar.healthRegenerator(counter++);
+            System.out.println("1");
+        }
+
+    }
+
+    //This is a private helper method to read in all of the locations in a text file and parse them to ingame Location objects.
+    private HashMap<String,Location> populateLocations(){
```

Comment:
Can you try to get the latest from dev again? This looks like the old implementation for the populateLocations method.